mmv1/third_party/terraform/tests/resource_compute_shared_reservation_update_test.go

```diff
@@ -0,0 +1,232 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeReservationDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeReservation_sharedReservation_basic(context),
+			},
+			{
+				ResourceName:            "google_compute_reservation.gce_reservation",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"zone", "share_settings"},
+			},
+			{
+				Config: testAccComputeReservation_sharedReservation_update(context),
+			},
+			{
+				ResourceName:            "google_compute_reservation.gce_reservation",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"zone", "share_settings"},
+			},
+			{
+				Config: testAccComputeReservation_sharedReservation_basic(context),
+			},
+			{
+				ResourceName:            "google_compute_reservation.gce_reservation",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"zone", "share_settings"},
+			},
+		},
+	})
+}
+
+func testAccComputeReservation_sharedReservation_basic(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_project" "owner_project" {
+  project_id      = "tf-test%{random_suffix}"
+  name            = "tf-test%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+
+
+resource "google_project_service" "compute" {
+  project = google_project.owner_project.project_id
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_project" "guest_project" {
+  project_id      = "tf-test-2%{random_suffix}"
+  name            = "tf-test-2%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+
+resource "google_project" "guest_project_second" {
+  project_id      = "tf-test-3%{random_suffix}"
+  name            = "tf-test-3%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+resource "google_project" "guest_project_third" {
+  project_id      = "tf-test-4%{random_suffix}"
+  name            = "tf-test-4%{random_suffix}"
+  org_id          = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+resource "google_project_service" "compute_second_project" {
+  project = google_project.guest_project.project_id
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+resource "google_project_service" "compute_third_project" {
+  project = google_project.guest_project_second.project_id
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+resource "google_project_service" "compute_fourth_project" {
+  project = google_project.guest_project_third.project_id
+  service = "compute.googleapis.com"
+  disable_on_destroy = false
+}
+
+resource "google_organization_policy" "shared_reservation_org_policy" {
```

Comment:
@rileykarson  it is the place We are applying sharedReservationOrgPolicy it used to be here from previous PR. I mean it is required feature to create sharedReservation and I have not changed it.